src/filesystem.c

```diff
@@ -275,6 +277,91 @@ rcutils_get_file_size(const char * file_path)
   return rc == 0 ? (size_t)(stat_buffer.st_size) : 0;
 }
 
+rcutils_ret_t
+rcutils_list_directory(
+  const char * directory_path,
+  rcutils_allocator_t allocator,
+  rcutils_string_array_t * string_array)
+{
+  RCUTILS_CHECK_FOR_NULL_WITH_MSG(
+    directory_path, "directory_path is null", return RCUTILS_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_FOR_NULL_WITH_MSG(
```

Comment:
rcutils_string_array_init also does this check.

Reply:
That's fair. To be pedantic, the argument name could be different in that function so the error message might not be correct, but it might be worth the performance since this is likely to be a rare error case.